### PR TITLE
Signet support

### PR DESCRIPTION
**Pull request by Jasonvdb**
- Signet support
- Added `currencyString()` func to Kotlin to match iOS and return a string in the invoice object
- Small refactor to align kotlin and swift

### CODE REVIEW

File: lib/android/src/main/java/com/reactnativeldk/Helpers.kt [hunk 1]
```diff
@@ -376,9 +376,21 @@ fun UserConfig.mergeWithMap(map: ReadableMap): UserConfig {
...
         else -> Pair(Network.LDKNetwork_Bitcoin, Currency.LDKCurrency_Bitcoin)
     }
+}
+
+fun currencyString(currency: Currency): String {
+    return when (currency) {
+        Currency.LDKCurrency_Bitcoin -> "Bitcoin"
+        Currency.LDKCurrency_BitcoinTestnet -> "BitcoinTestnet"
+        Currency.LDKCurrency_Regtest -> "Regtest"
+        Currency.LDKCurrency_Signet -> "Simnet"
```
Comment (coreyphillips) [inline lib/android/src/main/java/com/reactnativeldk/Helpers.kt:392]: Duplicate.

**Review by coreyphillips — CHANGES_REQUESTED** [review]

**Comment by bennyhodl** [comment]
`lib/src/utils/helpers.ts`:
`startParamCheck()` needs to have signet

`example/utils/helpers.ts`:
`ldkNetwork()` needs to have signet 👍

**Comment by bennyhodl** [comment]
All looks good when I added the changes above. I used https://github.com/synonymdev/react-native-ldk/pull/135 to test signet on [mutiny net](https://mutinynet.com).

Did find that bitcoinjs does not support signet. So I was only able to test adding a peer and opening a channel. Application crashes on get script pubkey history.

**Comment by bennyhodl** [comment]
Getting an error after the channel is confirmed in the Native setTxConfirmed. All info passed into looks correct

```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Io(Error { kind: UnexpectedEof, message: "failed to fill whole buffer" })', src/c_types/mod.rs:438:116
stack backtrace:
```

**Comment by Jasonvdb** [comment]
@bennyhodl maybe try this? https://github.com/bitcoinjs/bitcoinjs-lib/issues/1820#issuecomment-1193400719

**Comment by bennyhodl** [comment]
Params for `NativeLDK.setTxConfirmed()`

```
{
"header":"00000020139124db0d9d6146337157757b8312ea5356be78657ce2f49048a4079e000000cb2185d2859fdb659b6b15095f525ec217b451dbfaee47531d51f3072cf6cd988bbc9c64ae77031e6c6c0c00", 
"height": 195433, 
"transaction": "87e7e6c3012e8b2ee49aa4006dd39a40b8ffea9fb29323a2ede71a85fa522f7e", "vout": [{"hex": "00207776e80733beeeab67f22f3f8dbf6dd267c31a338c02181003babc92356a092e", "n": 0, "value": 1000000}, {"hex": "51204e13bbdc2ccbf8b6e9db49308d00bf717220256bb622e99b259a523c768f7902", "n": 1, "value": 7999690}]}
```

transaction: https://mutinynet.com/tx/87e7e6c3012e8b2ee49aa4006dd39a40b8ffea9fb29323a2ede71a85fa522f7e

block: https://mutinynet.com/block/000001fa4c271b5b335782836949d67c7a4d250b94eb51944407c42b8fe9be6f

**Comment by Jasonvdb** [comment]
> 



> {
> "header":"00000020139124db0d9d6146337157757b8312ea5356be78657ce2f49048a4079e000000cb2185d2859fdb659b6b15095f525ec217b451dbfaee47531d51f3072cf6cd988bbc9c64ae77031e6c6c0c00", 
> "height": 195433, 
> "transaction": "87e7e6c3012e8b2ee49aa4006dd39a40b8ffea9fb29323a2ede71a85fa522f7e", "vout": [{"hex": "00207776e80733beeeab67f22f3f8dbf6dd267c31a338c02181003babc92356a092e", "n": 0, "value": 1000000}, {"hex": "51204e13bbdc2ccbf8b6e9db49308d00bf717220256bb622e99b259a523c768f7902", "n": 1, "value": 7999690}]
> }

Looks like the wrong format and native can't parse it. Are you calling `NativeLDK.setTxConfirmed` directly? If you are rather call `ldk.setTxConfirmed(req)` with req being this format:  https://github.com/synonymdev/react-native-ldk/blob/f0320aee9e5fd64717727b6792f556bcce51e698/lib/src/utils/types.ts#L236

**Comment by Jasonvdb** [comment]
Updated startup param check [here](https://github.com/synonymdev/react-native-ldk/pull/138/commits/dafea84d2281f1cda02db00a6b00a67016796085)

**Comment by bennyhodl** [comment]
> > 
> Looks like the wrong format and native can't parse it. Are you calling `NativeLDK.setTxConfirmed` directly? If you are rather call `ldk.setTxConfirmed(req)` with req being this format:
> 
> https://github.com/synonymdev/react-native-ldk/blob/f0320aee9e5fd64717727b6792f556bcce51e698/lib/src/utils/types.ts#L236

ah i gave the wrong output.
```
{"header": "00000020139124db0d9d6146337157757b8312ea5356be78657ce2f49048a4079e000000cb2185d2859fdb659b6b15095f525ec217b451dbfaee47531d51f3072cf6cd988bbc9c64ae77031e6c6c0c00", 
"height": 195433, 
"txData": [{"pos": 1, "transaction": "87e7e6c3012e8b2ee49aa4006dd39a40b8ffea9fb29323a2ede71a85fa522f7e"}]}
```

**Comment by coreyphillips** [comment]
Created PR [here](https://github.com/synonymdev/react-native-ldk/pull/140) with some updates that add signet to the networks object.

**Review by coreyphillips — DISMISSED** [review]

**Pull request merged**